plugin/storage/es/factory.go

```diff
@@ -236,8 +266,26 @@ var _ io.Closer = (*Factory)(nil)
 
 // Close closes the resources held by the factory
 func (f *Factory) Close() error {
+	var errs []error
+	for _, w := range f.watchers {
+		errs = append(errs, w.Close())
+	}
 	if cfg := f.Options.Get(archiveNamespace); cfg != nil {
 		cfg.TLS.Close()
 	}
-	return f.Options.GetPrimary().TLS.Close()
+	errs = append(errs, f.Options.GetPrimary().TLS.Close())
+	return errors.Join(errs...)
+}
+
+func (f *Factory) onPasswordChange(config config.ClientBuilder, esClient es.Client) {
+	b, err := os.ReadFile(filepath.Clean(config.GetPasswordFilePath()))
+	if err == nil {
+		newPassword := strings.TrimRight(string(b), "\r\n")
+		config.SetPassword(newPassword)
```

Comment:
why do we need to set pwd on the config?

Reply:
Now I think we don't.

Let's see if I understand correctly:
- Users can set password in `config.Password` and/or `config.PasswordFilePath`
- The password from `config.PasswordFilePath` will override `config.Password`
- Later if `config.PasswordFilePath` is empty or deleted, we'll use `config.Password` instead

Is that correct?

Reply:
Only-once is a feasible implementation of the PasswordFilePath flag, but the point of watching the file for updates is to detect if the password is changed and begin using the new value. Same with token file.

Reply:
Yeah I understand `PasswordFilePath` needs to be watched. I already did it in the implementation.

What I'm discussing about is the (maybe rare) case when users set password in both `PasswordFilePath` and `Password`. In this case, the password from `PasswordFilePath` should override `Password`, right?

And later, in case `PasswordFilePath` is removed or empty-content, what would we do?

Initially, in the code that you reviewed, I called `SetPassword` to make sure `Password` and `PasswordFilePath` always had the same non-empty password. And in case `PasswordFilePath` is removed or empty-content, we'd use the latest non-empty password stored in `Password`. But now I'm not sure if it's a good idea.

Reply:
We should not allow setting both string pwd and file name, it's a configuration error that can halt the startup.

If the watched file becomes empty, our default behavior in all other components that watch files is to use the previously read value, and log a message informing the operator about that.